pubsublite/consumer.go

```diff
@@ -178,8 +188,20 @@ func (c *Consumer) Run(ctx context.Context) error {
 		}
 		batch := model.Batch{event}
 		ctx = queuecontext.WithMetadata(ctx, msg.Attributes)
-		if err := c.cfg.Processor.ProcessBatch(ctx, &batch); err != nil {
-			defer msg.Nack()
+		var err error
+		switch c.cfg.Delivery {
+		case apmqueue.AtMostOnceDeliveryType:
+			msg.Ack()
+		case apmqueue.AtLeastOnceDeliveryType:
+			defer func() {
+				if err != nil {
+					defer msg.Nack()
+				} else {
+					defer msg.Ack()
```

Comment:
```suggestion
					msg.Nack()
				} else {
					msg.Ack()
```

Reply:
if we want at least once delivery, then we should acknowledge **after** the message has been processed, not before, right? Same for Nack.

Reply:
oh never mind, double defer 🤦🏼‍♂️